google-storage/src/google-content-provider.ts

```diff
@@ -0,0 +1,216 @@
+      responseType: "json"
+    };
+  }
+  function createSuccessAjaxResponseForDeleteFile(responseDelete : any): AjaxResponse {
+    return {
+      originalEvent: {},
+      xhr: {
+		name: responseDelete.request.href,
+		headers: responseDelete.headers,
+	},
+      request: {},
+	  status: 204,
+      response: responseDelete.statusMessage,
+      responseText: JSON.stringify(responseDelete.metadata),
+      responseType: "json"
+    };
+  }
+  
+  function createErrorAjaxResponse(status: number, error: any): AjaxResponse {
+    return {
+      originalEvent: {},
+      xhr: {},
+      request: {},
+      status,
+      response: error,
+      responseText: JSON.stringify(error),
+      responseType: "json"
+    };
+  }
+//import service account details, bucketname, filename and other details from your configuration file
+/**	
+Get metadata of the file from the bucket
+* @param storage
+* @param bucketName
+* @param fileName
+* @returns An Observable with the response
+*/
+export function get(storage: any, bucketName: string, fileName: string) : Observable < AjaxResponse > {
+	const bucket = storage.bucket(bucketName);
+	const file = bucket.file(fileName);
+	
+	   var response= from(file.get()).pipe(
+        map((result : any) => {
+            return createSuccessAjaxResponse(result[0]);
+		}),
+		catchError(error => of(createErrorAjaxResponse(404, error)))
+	);
+	response.subscribe(response => console.log(response));
+	return response;
+}
+/**
+ * Updates a file.
+ * @param storage
+ * @param bucketName
+ * @returns An Observable with the response
+ */
+export function update(storage: any, bucketName: string, fileName: string): Observable < AjaxResponse > {
+	throw new Error("Not supported by Google API");
+}
+/**
+Uploads the file to the bucket
+* @param storage
+* @param bucketName
+* @param filePathLocal
+* @returns An Observable with the response
+*/
+export function create(storage: any, bucketName: string, filePathLocal: string): Observable < AjaxResponse > {
+	// Uploads a local file to the bucket
+	const bucket = storage.bucket(bucketName);
+	var response= from(bucket.upload(filePathLocal, {
+		// Support for HTTP requests made with `Accept-Encoding: gzip`
+		gzip: true,
+		metadata: {
+			cacheControl: 'no-cache',
+		},
+	})).pipe(
+        map((result : any) => {
+            return createSuccessAjaxResponse(result[0]);
+		}),
+		catchError(error => of(createErrorAjaxResponse(404, error)))
+	);
+	response.subscribe(response => console.log(response));
+	console.log(`${filePathLocal} uploaded to ${bucketName}.`);
+	return response;
+}
+/**
+Deletes the file from the bucket
+* @param storage
+* @param bucketName
+* @param fileName
+* @returns An Observable with the response
+*/
+export function save(storage: any, bucketName: string, fileName: string): Observable < AjaxResponse > {
+	const file = storage.bucket(bucketName).file(fileName);
+	const contents = 'This is the updated contents of the file.';
```

Comment:
This method needs to take the contents as a parameter (or more likely read them from the store).